src/main/java/frc/robot/KrakenSwerveModule.java

```diff
@@ -0,0 +1,263 @@
+    }
+
+    /**
+     * Set the Rotation Motor Speed with no PID
+     * @param speed from -1.0 to 1.0
+     */
+    public void setTransMotorRaw(double speed) {
+        mTransMotor.set(speed);
+    }
+
+    /**
+     * Set the Rotation Motor Speed with no PID
+     * @param speed from -1.0 to 1.0
+     */
+    public void setRotMotorRaw(double speed) {
+        mRotMotor.set(speed);
+    }
+
+    /**
+     * @return Returns rotation position in radians
+     */
+    private double getRotPositionRaw() {
+        return mRotEncoder.getAbsolutePosition().getValueAsDouble()*2*Math.PI;
+    }
+
+    /**
+     * @return Returns number rotations of translation motor BEFORE GEAR RATIO
+     */
+    private double getTransPositionRaw() {
+        return mTransMotor.getPosition().getValueAsDouble();
+    }
+
+    /**
+     * 
+     * @return Returns rotation in RADIANS of rotation motor AFTER GEAR RATIO
+     */
+    public double getRotPosition() {
+        return getRotPositionRaw() * KrakenSwerveConstants.ABS_ENC_GEAR_RATIO_ROT;
+    }
+
+    /**
+     * 
+     * @return Returns translation motor AFTER GEAR RATIO and Meters
+     */
+    public double getTransPosition() {
+        return getTransPositionRaw() * KrakenSwerveConstants.TRANS_GEAR_RATIO_ROT * KrakenSwerveConstants.WHEEL_CIRCUMFERENCE_METERS;
+    }
+
+    /**
+     * @return Velocity in Rotations per minute
+     */
+    private double getTransVelocityRaw() {
+        return mTransMotor.getVelocity().getValueAsDouble()/60;
+    }
+
+    public double getRotRelativePosition() {
+        return mRotRelativeEncoder.getPosition() * KrakenSwerveConstants.REL_ENC_GEAR_RATIO_ROT;
+    }
+
+    /**
+     * 
+     * @return Returns velocity of translation motor with conversion
+     */
+    public double getTransVelocity() {
+        return getTransVelocityRaw() * SwerveConstants.TRANS_RPM_TO_MPS;
+    }
+
+    /**
+     * 
+     * @return the total distance traveled by the module (Meters) and Rotation value (Rad) in the
+     *         form of a SwerveModulePostion object
+     * @see SwerveModulePosition
+     */
+    public SwerveModulePosition getModulePos() {
+        return new SwerveModulePosition(getTransPosition(), Rotation2d.fromRadians(getRotPosition()));
+    }
+
+    /**
+     * 
+     * @return a swerve module state object describing the current speed of the translation and
+     *         rotation motors
+     * @see SwerveModuleState
+     */
+    public SwerveModuleState getState() {
+        return new SwerveModuleState(getTransVelocity(), Rotation2d.fromRadians(getRotPosition()));
+    }
+
+    /**
+     * Sets the motor speeds passed into constructor
+     * 
+     * @param desiredState takes in SwerveModule state
+     * @see SwerveModuleState
+     */
+    public void setDesiredState(SwerveModuleState desiredState) {
+
+        // Stops returning to original rotation
+        if (Math.abs(desiredState.speedMetersPerSecond) < 0.001) {
```

Comment:
0.001 is your error constant right, pull this up as a CONST easier to update globally if the system is noisy.